source/AI.cpp

```diff
@@ -2595,6 +2600,22 @@ void AI::DoScatter(Ship &ship, Command &command)
 
 
 
+// Rove; i. e. wander aimlessly around the system
+void AI::Rove(Ship &ship, Command &command) 
+{
+	// If this is a carried ship (like a fighter), or if the ship is disabled or destroyed, don't rove.
+	if(ship.CanBeCarried() || ship.IsDisabled() || ship.IsDestroyed())
+		return;
```

Comment:
This is redundant with code that must be executed before reaching this point; a destroyed or disabled ship will never get past line 570 in Step. And this check to see if the ship can be carried should probably be done before figuring out if we should rove, as otherwise a carried ship might end up receiving no commands, as it enters the Rove function but gets kicked right back out and doesn't get sent anywhere else (at least how you currently have it set up).